{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def three_end_mapping(samgen,GFFgen_filename):\n",
    "    from BCBio import GFF\n",
    "    from Bio import Seq\n",
    "    outputdata1={}\n",
    "    outputdata2={}\n",
    "    GFFgen=GFF.parse(GFFgen_filename)\n",
    "    for sequence in GFFgen:\n",
    "        outputdata1[sequence.id]=[0 for x in range(len(sequence))]\n",
    "        outputdata2[sequence.id]=[0 for x in range(len(sequence))]\n",
    "        for read in samgen:\n",
    "            if read[0][0] == '@':   # Ignore header lines.\n",
    "                continue\n",
    "            if read[1] == '4':      # A bowtie mismatch. These are the same as those previously saved in the files bowtie output as reads without matches and reads exceeding the -m limit. No need to write as fastq.\n",
    "                continue            \n",
    "            chrom = read[2]             # chromosome identified for read in bowtie\n",
    "            readid = read[0]            # read id\n",
    "            startp = int(read[3]) - 1    # start position. Need to subtract 1 since genomic sequence starts at 1, but list structures for read summing start at 0.\n",
    "            seq = Seq.Seq(read[9])      # sequence of the read\n",
    "            length = len(seq)           # length of read\n",
    "            if (read[1] == '16'):\n",
    "                start = startp\n",
    "                outputdata2[chrom][start] += 1\n",
    "            if (read[1] == '0'):\n",
    "                start = startp + length - 1\n",
    "                outputdata1[chrom][start] += 1\n",
    "    return [outputdata1,outputdata2]  \n",
    "\n",
    "def readcounts_to_rpm(readcounts):\n",
    "    for chrom in readcounts.keys():\n",
    "        totalread = sum(readcounts[chrom])\n",
    "        for position in range(len(readcounts[chrom])):\n",
    "            readcounts[chrom][position]/=float(totalread)\n",
    "            readcounts[chrom][position]*=1E6\n",
    "\n",
    "# makes density file \n",
    "# the last letter \"f\" in writecountsf means float\n",
    "def writecountsf(rpm,filepath):  #Resultlist it is the list of counts, filestring is the file prefix for each chr to be written.\n",
    "    import struct\n",
    "    f2=open(filepath+\"keys\",\"w\")\n",
    "    for chrom in rpm.keys():\n",
    "        f=open(filepath+\"genome\",\"wb\")\n",
    "        # used to be (filepath+chrom,\"wb\") but server fail to show a file with a name containing \"|\"\n",
    "        # rename \"genome\" to real genome name manually \n",
    "        for position in rpm[chrom]:\n",
    "            f.write(struct.pack(\"f\",position))\n",
    "            # struct.pack to make binary based on the format indicated \n",
    "            # here, the indication of format is \"f\", which means float\n",
    "        f.close()\n",
    "        f2.write(chrom+\"\\n\")\n",
    "    f2.close()\n",
    "\n",
    "# makes wig file\n",
    "def countstowig(rpm,filestring):\n",
    "    import random\n",
    "    f=open(filestring+\".wig\",\"w\")\n",
    "    filestring=filestring.partition(\"_\")[0][-3:]\n",
    "\n",
    "    random.seed(filestring)\n",
    "    c1=random.randint(0,255)\n",
    "    random.seed(filestring+\"xxx\")\n",
    "    c2=random.randint(0,255)\n",
    "    random.seed(filestring+\"000\")\n",
    "    c3=random.randint(0,255)\n",
    "\n",
    "    f.write(\"track name=tracklabel viewLimits=-5:5 color=\"+str(c1)+','+str(c2)+','+str(c3)+\"\\n\")\n",
    "    for chrom in rpm.keys():\n",
    "        if chrom[0:3]=='chr':\n",
    "            f.write(\"fixedStep  chrom=\"+chrom+\"  start=1  step=1\\n\")\n",
    "        else:\n",
    "            f.write(\"fixedStep  chrom=\\\"\"+chrom+\"\\\"  start=1  step=1\\n\")\t# Note the extra '' around chrom added for E coli. Should be fine for yeast. Actually it's not. So new code here to detect that.\n",
    "\n",
    "\n",
    "        for position in rpm[chrom]:\n",
    "            f.write(str(position)+\"\\n\")\n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fastq_path = '/home/kazuki/^profilingdata/20160420/'\n",
    "filepath = '/home/kazuki/^profilingdata/20160420_reanalyze/'\n",
    "files = ['ks'+str(x) for x in range(5,23)]\n",
    "bowtie_input_extension = '-trimmed.fastq'\n",
    "mapping = '_3map'\n",
    "gene_type = '_gene'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skewer for ks5\n",
      "skewer for ks6\n",
      "skewer for ks7\n",
      "skewer for ks8\n",
      "skewer for ks9\n",
      "skewer for ks10\n",
      "skewer for ks11\n",
      "skewer for ks12\n",
      "skewer for ks13\n",
      "skewer for ks14\n",
      "skewer for ks15\n",
      "skewer for ks16\n",
      "skewer for ks17\n",
      "skewer for ks18\n",
      "skewer for ks19\n",
      "skewer for ks20\n",
      "skewer for ks21\n",
      "skewer for ks22\n",
      "\n",
      "\n",
      "2019-06-03 09:25:25.448889\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:25:25 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:26:04 2019\u001b[0;32m >> done\u001b[0m (39.127s)\n",
      "18495146 reads processed; of these:\n",
      "    6792 ( 0.04%) reads filtered out by quality control\n",
      " 3368864 (18.21%) short reads filtered out after trimming by size control\n",
      "  698639 ( 3.78%) empty reads filtered out after trimming by size control\n",
      "14420851 (77.97%) reads available; of these:\n",
      "13719122 (95.13%) trimmed reads available after processing\n",
      "  701729 ( 4.87%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks5-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:26:04 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:26:25 2019\u001b[0;32m >> done\u001b[0m (21.167s)\n",
      "9760065 reads processed; of these:\n",
      "   3053 ( 0.03%) reads filtered out by quality control\n",
      "1827359 (18.72%) short reads filtered out after trimming by size control\n",
      " 562038 ( 5.76%) empty reads filtered out after trimming by size control\n",
      "7367615 (75.49%) reads available; of these:\n",
      "6975711 (94.68%) trimmed reads available after processing\n",
      " 391904 ( 5.32%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks6-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:26:25 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:26:52 2019\u001b[0;32m >> done\u001b[0m (26.943s)\n",
      "12609432 reads processed; of these:\n",
      "    5197 ( 0.04%) reads filtered out by quality control\n",
      " 2303504 (18.27%) short reads filtered out after trimming by size control\n",
      " 1614786 (12.81%) empty reads filtered out after trimming by size control\n",
      " 8685945 (68.88%) reads available; of these:\n",
      " 8082723 (93.06%) trimmed reads available after processing\n",
      "  603222 ( 6.94%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks7-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:26:52 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:27:04 2019\u001b[0;32m >> done\u001b[0m (11.437s)\n",
      "5059808 reads processed; of these:\n",
      "   2009 ( 0.04%) reads filtered out by quality control\n",
      "1043260 (20.62%) short reads filtered out after trimming by size control\n",
      " 373471 ( 7.38%) empty reads filtered out after trimming by size control\n",
      "3641068 (71.96%) reads available; of these:\n",
      "3413788 (93.76%) trimmed reads available after processing\n",
      " 227280 ( 6.24%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks8-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:27:04 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:27:39 2019\u001b[0;32m >> done\u001b[0m (35.372s)\n",
      "16621487 reads processed; of these:\n",
      "    6432 ( 0.04%) reads filtered out by quality control\n",
      " 3149571 (18.95%) short reads filtered out after trimming by size control\n",
      " 1157505 ( 6.96%) empty reads filtered out after trimming by size control\n",
      "12307979 (74.05%) reads available; of these:\n",
      "11617910 (94.39%) trimmed reads available after processing\n",
      "  690069 ( 5.61%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks9-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:27:39 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:28:16 2019\u001b[0;32m >> done\u001b[0m (36.937s)\n",
      "17381575 reads processed; of these:\n",
      "    5710 ( 0.03%) reads filtered out by quality control\n",
      " 4065895 (23.39%) short reads filtered out after trimming by size control\n",
      "  893984 ( 5.14%) empty reads filtered out after trimming by size control\n",
      "12415986 (71.43%) reads available; of these:\n",
      "11750195 (94.64%) trimmed reads available after processing\n",
      "  665791 ( 5.36%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks10-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:28:16 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:29:38 2019\u001b[0;32m >> done\u001b[0m (82.373s)\n",
      "41304153 reads processed; of these:\n",
      "   13154 ( 0.03%) reads filtered out by quality control\n",
      "17156550 (41.54%) short reads filtered out after trimming by size control\n",
      " 6194830 (15.00%) empty reads filtered out after trimming by size control\n",
      "17939619 (43.43%) reads available; of these:\n",
      "15906256 (88.67%) trimmed reads available after processing\n",
      " 2033363 (11.33%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks11-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:29:38 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:30:41 2019\u001b[0;32m >> done\u001b[0m (62.921s)\n",
      "30678414 reads processed; of these:\n",
      "   10335 ( 0.03%) reads filtered out by quality control\n",
      " 9860359 (32.14%) short reads filtered out after trimming by size control\n",
      " 3093234 (10.08%) empty reads filtered out after trimming by size control\n",
      "17714486 (57.74%) reads available; of these:\n",
      "16354514 (92.32%) trimmed reads available after processing\n",
      " 1359972 ( 7.68%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks12-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:30:41 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:31:33 2019\u001b[0;32m >> done\u001b[0m (51.535s)\n",
      "24678394 reads processed; of these:\n",
      "    8437 ( 0.03%) reads filtered out by quality control\n",
      " 6583667 (26.68%) short reads filtered out after trimming by size control\n",
      " 2230908 ( 9.04%) empty reads filtered out after trimming by size control\n",
      "15855382 (64.25%) reads available; of these:\n",
      "14830262 (93.53%) trimmed reads available after processing\n",
      " 1025120 ( 6.47%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks13-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:31:33 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:31:59 2019\u001b[0;32m >> done\u001b[0m (26.253s)\n",
      "12378224 reads processed; of these:\n",
      "    4213 ( 0.03%) reads filtered out by quality control\n",
      " 3556450 (28.73%) short reads filtered out after trimming by size control\n",
      "  986644 ( 7.97%) empty reads filtered out after trimming by size control\n",
      " 7830917 (63.26%) reads available; of these:\n",
      " 7253761 (92.63%) trimmed reads available after processing\n",
      "  577156 ( 7.37%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks14-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:31:59 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:32:37 2019\u001b[0;32m >> done\u001b[0m (37.560s)\n",
      "17780946 reads processed; of these:\n",
      "    6189 ( 0.03%) reads filtered out by quality control\n",
      " 4904953 (27.59%) short reads filtered out after trimming by size control\n",
      " 1489916 ( 8.38%) empty reads filtered out after trimming by size control\n",
      "11379888 (64.00%) reads available; of these:\n",
      "10631422 (93.42%) trimmed reads available after processing\n",
      "  748466 ( 6.58%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks15-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:32:37 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:33:27 2019\u001b[0;32m >> done\u001b[0m (50.087s)\n",
      "23717958 reads processed; of these:\n",
      "    8159 ( 0.03%) reads filtered out by quality control\n",
      " 7738766 (32.63%) short reads filtered out after trimming by size control\n",
      "  896089 ( 3.78%) empty reads filtered out after trimming by size control\n",
      "15074944 (63.56%) reads available; of these:\n",
      "14182803 (94.08%) trimmed reads available after processing\n",
      "  892141 ( 5.92%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks16-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:33:27 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:34:04 2019\u001b[0;32m >> done\u001b[0m (37.531s)\n",
      "17213917 reads processed; of these:\n",
      "    4873 ( 0.03%) reads filtered out by quality control\n",
      " 4158526 (24.16%) short reads filtered out after trimming by size control\n",
      "   10130 ( 0.06%) empty reads filtered out after trimming by size control\n",
      "13040388 (75.75%) reads available; of these:\n",
      "12304152 (94.35%) trimmed reads available after processing\n",
      "  736236 ( 5.65%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks17-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:34:04 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:34:36 2019\u001b[0;32m >> done\u001b[0m (31.950s)\n",
      "14684178 reads processed; of these:\n",
      "    3970 ( 0.03%) reads filtered out by quality control\n",
      " 3464770 (23.60%) short reads filtered out after trimming by size control\n",
      "   12774 ( 0.09%) empty reads filtered out after trimming by size control\n",
      "11202664 (76.29%) reads available; of these:\n",
      "10546912 (94.15%) trimmed reads available after processing\n",
      "  655752 ( 5.85%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks18-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:34:36 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:35:25 2019\u001b[0;32m >> done\u001b[0m (49.220s)\n",
      "22985988 reads processed; of these:\n",
      "    6469 ( 0.03%) reads filtered out by quality control\n",
      " 6133679 (26.68%) short reads filtered out after trimming by size control\n",
      "   35248 ( 0.15%) empty reads filtered out after trimming by size control\n",
      "16810592 (73.13%) reads available; of these:\n",
      "15748798 (93.68%) trimmed reads available after processing\n",
      " 1061794 ( 6.32%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks19-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:35:25 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:36:03 2019\u001b[0;32m >> done\u001b[0m (37.800s)\n",
      "17526813 reads processed; of these:\n",
      "    5118 ( 0.03%) reads filtered out by quality control\n",
      " 6076220 (34.67%) short reads filtered out after trimming by size control\n",
      "   18968 ( 0.11%) empty reads filtered out after trimming by size control\n",
      "11426507 (65.19%) reads available; of these:\n",
      "10666428 (93.35%) trimmed reads available after processing\n",
      "  760079 ( 6.65%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks20-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:36:03 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:36:41 2019\u001b[0;32m >> done\u001b[0m (37.496s)\n",
      "18357760 reads processed; of these:\n",
      "    5162 ( 0.03%) reads filtered out by quality control\n",
      "10135527 (55.21%) short reads filtered out after trimming by size control\n",
      "  114160 ( 0.62%) empty reads filtered out after trimming by size control\n",
      " 8102911 (44.14%) reads available; of these:\n",
      " 7448056 (91.92%) trimmed reads available after processing\n",
      "  654855 ( 8.08%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks21-trimmed.log\".\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tCTGTAGGCACCATCAAT\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- mean quality threshold (-Q):\t\t10\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- minimum overlap length for adapter detection (-k):\t3\n",
      "Mon Jun  3 09:36:41 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun  3 09:37:13 2019\u001b[0;32m >> done\u001b[0m (31.962s)\n",
      "15739277 reads processed; of these:\n",
      "    4674 ( 0.03%) reads filtered out by quality control\n",
      " 9072391 (57.64%) short reads filtered out after trimming by size control\n",
      "  102335 ( 0.65%) empty reads filtered out after trimming by size control\n",
      " 6559877 (41.68%) reads available; of these:\n",
      " 6025409 (91.85%) trimmed reads available after processing\n",
      "  534468 ( 8.15%) untrimmed reads available after processing\n",
      "log has been saved to \"/home/kazuki/^profilingdata/20160420_reanalyze/ks22-trimmed.log\".\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "from datetime import datetime\n",
    "import os\n",
    "\n",
    "skewer = '/home/kazuki/^data_analysis/analtools/skewer022/skewer-0.2.2-linux-x86_64'\n",
    "option = '-Q 10'\n",
    "adapter = 'CTGTAGGCACCATCAAT'\n",
    "\n",
    "log_file = filepath+'log_skewer.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "present = str(datetime.now())\n",
    "logs.write(present+'\\n')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "\n",
    "for f in files:\n",
    "    print 'skewer for ' + f\n",
    "    file_path = filepath+f+'.fastq'\n",
    "    \n",
    "    command = '%s %s -x %s %s >> %s' % (\n",
    "        skewer,\n",
    "        option,\n",
    "        adapter,\n",
    "        file_path,\n",
    "        log_file\n",
    "        ) \n",
    "    \n",
    "    os.system(command)\n",
    "    \n",
    "print \n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks5\n",
      "ks6\n",
      "ks7\n",
      "ks8\n",
      "ks9\n",
      "ks10\n",
      "ks11\n",
      "ks12\n",
      "ks13\n",
      "ks14\n",
      "ks15\n",
      "ks16\n",
      "ks17\n",
      "ks18\n",
      "ks19\n",
      "ks20\n",
      "ks21\n",
      "ks22\n",
      "\n",
      "\n",
      "2019-06-03 09:37:13.393750\n",
      "\n",
      "\n",
      "ks5\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks5_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks5_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks5_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks5-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks5_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14420851\n",
      "# reads with at least one reported alignment: 58816 (0.41%)\n",
      "# reads that failed to align: 14352580 (99.53%)\n",
      "# reads with alignments suppressed due to -m: 9455 (0.07%)\n",
      "Reported 58816 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks5_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks5_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks5_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks5_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks5_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14352580\n",
      "# reads with at least one reported alignment: 1665557 (11.60%)\n",
      "# reads that failed to align: 7885336 (54.94%)\n",
      "# reads with alignments suppressed due to -m: 4801687 (33.46%)\n",
      "Reported 1665557 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks5_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks5_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks5_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks5_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks5_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7885336\n",
      "# reads with at least one reported alignment: 10350 (0.13%)\n",
      "# reads that failed to align: 6574712 (83.38%)\n",
      "# reads with alignments suppressed due to -m: 1300274 (16.49%)\n",
      "Reported 10350 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks5_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks5_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks5_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks5_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks5_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6574712\n",
      "# reads with at least one reported alignment: 5236460 (79.65%)\n",
      "# reads that failed to align: 1207162 (18.36%)\n",
      "# reads with alignments suppressed due to -m: 131090 (1.99%)\n",
      "Reported 5236460 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks6\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks6_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks6_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks6_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks6-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks6_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7367615\n",
      "# reads with at least one reported alignment: 11889 (0.16%)\n",
      "# reads that failed to align: 7353758 (99.81%)\n",
      "# reads with alignments suppressed due to -m: 1968 (0.03%)\n",
      "Reported 11889 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks6_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks6_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks6_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks6_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks6_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7353758\n",
      "# reads with at least one reported alignment: 827529 (11.25%)\n",
      "# reads that failed to align: 3973260 (54.03%)\n",
      "# reads with alignments suppressed due to -m: 2552969 (34.72%)\n",
      "Reported 827529 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks6_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks6_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks6_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks6_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks6_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3973260\n",
      "# reads with at least one reported alignment: 5824 (0.15%)\n",
      "# reads that failed to align: 3446413 (86.74%)\n",
      "# reads with alignments suppressed due to -m: 521023 (13.11%)\n",
      "Reported 5824 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks6_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks6_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks6_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks6_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks6_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3446413\n",
      "# reads with at least one reported alignment: 2790468 (80.97%)\n",
      "# reads that failed to align: 586470 (17.02%)\n",
      "# reads with alignments suppressed due to -m: 69475 (2.02%)\n",
      "Reported 2790468 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks7\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks7_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks7_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks7_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks7-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks7_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 8685945\n",
      "# reads with at least one reported alignment: 19476 (0.22%)\n",
      "# reads that failed to align: 8663253 (99.74%)\n",
      "# reads with alignments suppressed due to -m: 3216 (0.04%)\n",
      "Reported 19476 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks7_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks7_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks7_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks7_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks7_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 8663253\n",
      "# reads with at least one reported alignment: 1136404 (13.12%)\n",
      "# reads that failed to align: 4415242 (50.97%)\n",
      "# reads with alignments suppressed due to -m: 3111607 (35.92%)\n",
      "Reported 1136404 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks7_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks7_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks7_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks7_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks7_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 4415242\n",
      "# reads with at least one reported alignment: 5944 (0.13%)\n",
      "# reads that failed to align: 3978334 (90.10%)\n",
      "# reads with alignments suppressed due to -m: 430964 (9.76%)\n",
      "Reported 5944 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks7_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks7_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks7_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks7_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks7_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3978334\n",
      "# reads with at least one reported alignment: 3104436 (78.03%)\n",
      "# reads that failed to align: 796803 (20.03%)\n",
      "# reads with alignments suppressed due to -m: 77095 (1.94%)\n",
      "Reported 3104436 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks8\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks8_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks8_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks8_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks8-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks8_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3641068\n",
      "# reads with at least one reported alignment: 7124 (0.20%)\n",
      "# reads that failed to align: 3632826 (99.77%)\n",
      "# reads with alignments suppressed due to -m: 1118 (0.03%)\n",
      "Reported 7124 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks8_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks8_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks8_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks8_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks8_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3632826\n",
      "# reads with at least one reported alignment: 639912 (17.61%)\n",
      "# reads that failed to align: 1540222 (42.40%)\n",
      "# reads with alignments suppressed due to -m: 1452692 (39.99%)\n",
      "Reported 639912 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks8_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks8_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks8_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks8_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks8_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 1540222\n",
      "# reads with at least one reported alignment: 2853 (0.19%)\n",
      "# reads that failed to align: 1305673 (84.77%)\n",
      "# reads with alignments suppressed due to -m: 231696 (15.04%)\n",
      "Reported 2853 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks8_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks8_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks8_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks8_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks8_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 1305673\n",
      "# reads with at least one reported alignment: 968560 (74.18%)\n",
      "# reads that failed to align: 315401 (24.16%)\n",
      "# reads with alignments suppressed due to -m: 21712 (1.66%)\n",
      "Reported 968560 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks9\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks9_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks9_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks9_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks9-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks9_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12307979\n",
      "# reads with at least one reported alignment: 28436 (0.23%)\n",
      "# reads that failed to align: 12274422 (99.73%)\n",
      "# reads with alignments suppressed due to -m: 5121 (0.04%)\n",
      "Reported 28436 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks9_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks9_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks9_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks9_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks9_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12274422\n",
      "# reads with at least one reported alignment: 1956650 (15.94%)\n",
      "# reads that failed to align: 4793495 (39.05%)\n",
      "# reads with alignments suppressed due to -m: 5524277 (45.01%)\n",
      "Reported 1956650 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks9_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks9_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks9_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks9_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks9_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 4793495\n",
      "# reads with at least one reported alignment: 9753 (0.20%)\n",
      "# reads that failed to align: 3907882 (81.52%)\n",
      "# reads with alignments suppressed due to -m: 875860 (18.27%)\n",
      "Reported 9753 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks9_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks9_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks9_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks9_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks9_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3907882\n",
      "# reads with at least one reported alignment: 2824859 (72.29%)\n",
      "# reads that failed to align: 1024884 (26.23%)\n",
      "# reads with alignments suppressed due to -m: 58139 (1.49%)\n",
      "Reported 2824859 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks10\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks10_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks10_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks10_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks10-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks10_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12415986\n",
      "# reads with at least one reported alignment: 50968 (0.41%)\n",
      "# reads that failed to align: 12357397 (99.53%)\n",
      "# reads with alignments suppressed due to -m: 7621 (0.06%)\n",
      "Reported 50968 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks10_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks10_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks10_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks10_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks10_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12357397\n",
      "# reads with at least one reported alignment: 2502949 (20.25%)\n",
      "# reads that failed to align: 3490938 (28.25%)\n",
      "# reads with alignments suppressed due to -m: 6363510 (51.50%)\n",
      "Reported 2502949 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks10_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks10_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks10_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks10_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks10_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3490938\n",
      "# reads with at least one reported alignment: 15012 (0.43%)\n",
      "# reads that failed to align: 2710482 (77.64%)\n",
      "# reads with alignments suppressed due to -m: 765444 (21.93%)\n",
      "Reported 15012 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks10_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks10_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks10_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks10_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks10_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 2710482\n",
      "# reads with at least one reported alignment: 1753276 (64.69%)\n",
      "# reads that failed to align: 917658 (33.86%)\n",
      "# reads with alignments suppressed due to -m: 39548 (1.46%)\n",
      "Reported 1753276 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks11\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks11_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks11_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks11_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks11-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks11_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 17939619\n",
      "# reads with at least one reported alignment: 581639 (3.24%)\n",
      "# reads that failed to align: 17293380 (96.40%)\n",
      "# reads with alignments suppressed due to -m: 64600 (0.36%)\n",
      "Reported 581639 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks11_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks11_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks11_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks11_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks11_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 17293380\n",
      "# reads with at least one reported alignment: 357587 (2.07%)\n",
      "# reads that failed to align: 16340345 (94.49%)\n",
      "# reads with alignments suppressed due to -m: 595448 (3.44%)\n",
      "Reported 357587 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks11_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks11_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks11_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks11_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks11_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 16340345\n",
      "# reads with at least one reported alignment: 30003 (0.18%)\n",
      "# reads that failed to align: 14666742 (89.76%)\n",
      "# reads with alignments suppressed due to -m: 1643600 (10.06%)\n",
      "Reported 30003 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks11_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks11_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks11_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks11_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks11_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14666742\n",
      "# reads with at least one reported alignment: 11686555 (79.68%)\n",
      "# reads that failed to align: 2567355 (17.50%)\n",
      "# reads with alignments suppressed due to -m: 412832 (2.81%)\n",
      "Reported 11686555 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks12\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks12_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks12_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks12_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks12-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks12_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 17714486\n",
      "# reads with at least one reported alignment: 101561 (0.57%)\n",
      "# reads that failed to align: 17597227 (99.34%)\n",
      "# reads with alignments suppressed due to -m: 15698 (0.09%)\n",
      "Reported 101561 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks12_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks12_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks12_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks12_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks12_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 17597227\n",
      "# reads with at least one reported alignment: 410495 (2.33%)\n",
      "# reads that failed to align: 16443969 (93.45%)\n",
      "# reads with alignments suppressed due to -m: 742763 (4.22%)\n",
      "Reported 410495 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks12_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks12_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks12_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks12_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks12_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 16443969\n",
      "# reads with at least one reported alignment: 33545 (0.20%)\n",
      "# reads that failed to align: 14420717 (87.70%)\n",
      "# reads with alignments suppressed due to -m: 1989707 (12.10%)\n",
      "Reported 33545 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks12_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks12_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks12_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks12_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks12_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14420717\n",
      "# reads with at least one reported alignment: 12052268 (83.58%)\n",
      "# reads that failed to align: 1950319 (13.52%)\n",
      "# reads with alignments suppressed due to -m: 418130 (2.90%)\n",
      "Reported 12052268 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks13\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks13_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks13_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks13_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks13-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks13_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 15855382\n",
      "# reads with at least one reported alignment: 56177 (0.35%)\n",
      "# reads that failed to align: 15790550 (99.59%)\n",
      "# reads with alignments suppressed due to -m: 8655 (0.05%)\n",
      "Reported 56177 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks13_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks13_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks13_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks13_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks13_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 15790550\n",
      "# reads with at least one reported alignment: 558134 (3.53%)\n",
      "# reads that failed to align: 14203012 (89.95%)\n",
      "# reads with alignments suppressed due to -m: 1029404 (6.52%)\n",
      "Reported 558134 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks13_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks13_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks13_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks13_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks13_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14203012\n",
      "# reads with at least one reported alignment: 16978 (0.12%)\n",
      "# reads that failed to align: 12335341 (86.85%)\n",
      "# reads with alignments suppressed due to -m: 1850693 (13.03%)\n",
      "Reported 16978 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks13_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks13_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks13_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks13_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks13_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12335341\n",
      "# reads with at least one reported alignment: 10489070 (85.03%)\n",
      "# reads that failed to align: 1400491 (11.35%)\n",
      "# reads with alignments suppressed due to -m: 445780 (3.61%)\n",
      "Reported 10489070 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks14\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks14_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks14_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks14_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks14-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks14_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7830917\n",
      "# reads with at least one reported alignment: 60581 (0.77%)\n",
      "# reads that failed to align: 7764900 (99.16%)\n",
      "# reads with alignments suppressed due to -m: 5436 (0.07%)\n",
      "Reported 60581 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks14_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks14_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks14_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks14_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks14_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7764900\n",
      "# reads with at least one reported alignment: 224571 (2.89%)\n",
      "# reads that failed to align: 7115503 (91.64%)\n",
      "# reads with alignments suppressed due to -m: 424826 (5.47%)\n",
      "Reported 224571 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks14_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks14_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks14_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks14_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks14_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 7115503\n",
      "# reads with at least one reported alignment: 8304 (0.12%)\n",
      "# reads that failed to align: 6424687 (90.29%)\n",
      "# reads with alignments suppressed due to -m: 682512 (9.59%)\n",
      "Reported 8304 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks14_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks14_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks14_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks14_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks14_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6424687\n",
      "# reads with at least one reported alignment: 5487440 (85.41%)\n",
      "# reads that failed to align: 769153 (11.97%)\n",
      "# reads with alignments suppressed due to -m: 168094 (2.62%)\n",
      "Reported 5487440 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks15\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks15_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks15_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks15_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks15-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks15_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11379888\n",
      "# reads with at least one reported alignment: 223276 (1.96%)\n",
      "# reads that failed to align: 11128758 (97.79%)\n",
      "# reads with alignments suppressed due to -m: 27854 (0.24%)\n",
      "Reported 223276 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks15_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks15_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks15_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks15_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks15_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11128758\n",
      "# reads with at least one reported alignment: 405183 (3.64%)\n",
      "# reads that failed to align: 9992672 (89.79%)\n",
      "# reads with alignments suppressed due to -m: 730903 (6.57%)\n",
      "Reported 405183 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks15_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks15_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks15_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks15_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks15_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 9992672\n",
      "# reads with at least one reported alignment: 15785 (0.16%)\n",
      "# reads that failed to align: 9159500 (91.66%)\n",
      "# reads with alignments suppressed due to -m: 817387 (8.18%)\n",
      "Reported 15785 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks15_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks15_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks15_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks15_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks15_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 9159500\n",
      "# reads with at least one reported alignment: 7981550 (87.14%)\n",
      "# reads that failed to align: 964181 (10.53%)\n",
      "# reads with alignments suppressed due to -m: 213769 (2.33%)\n",
      "Reported 7981550 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks16\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks16_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks16_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks16_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks16-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks16_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 15074944\n",
      "# reads with at least one reported alignment: 714220 (4.74%)\n",
      "# reads that failed to align: 14256865 (94.57%)\n",
      "# reads with alignments suppressed due to -m: 103859 (0.69%)\n",
      "Reported 714220 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks16_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks16_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks16_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks16_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks16_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 14256865\n",
      "# reads with at least one reported alignment: 514360 (3.61%)\n",
      "# reads that failed to align: 12904701 (90.52%)\n",
      "# reads with alignments suppressed due to -m: 837804 (5.88%)\n",
      "Reported 514360 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks16_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks16_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks16_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks16_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks16_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 12904701\n",
      "# reads with at least one reported alignment: 19604 (0.15%)\n",
      "# reads that failed to align: 11758641 (91.12%)\n",
      "# reads with alignments suppressed due to -m: 1126456 (8.73%)\n",
      "Reported 19604 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks16_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks16_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks16_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks16_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks16_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11758641\n",
      "# reads with at least one reported alignment: 10189904 (86.66%)\n",
      "# reads that failed to align: 1292016 (10.99%)\n",
      "# reads with alignments suppressed due to -m: 276721 (2.35%)\n",
      "Reported 10189904 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks17\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks17_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks17_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks17_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks17-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks17_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 13040388\n",
      "# reads with at least one reported alignment: 16057 (0.12%)\n",
      "# reads that failed to align: 13021708 (99.86%)\n",
      "# reads with alignments suppressed due to -m: 2623 (0.02%)\n",
      "Reported 16057 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks17_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks17_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks17_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks17_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks17_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 13021708\n",
      "# reads with at least one reported alignment: 2052694 (15.76%)\n",
      "# reads that failed to align: 5612214 (43.10%)\n",
      "# reads with alignments suppressed due to -m: 5356800 (41.14%)\n",
      "Reported 2052694 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks17_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks17_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks17_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks17_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks17_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 5612214\n",
      "# reads with at least one reported alignment: 141005 (2.51%)\n",
      "# reads that failed to align: 3846722 (68.54%)\n",
      "# reads with alignments suppressed due to -m: 1624487 (28.95%)\n",
      "Reported 141005 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks17_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks17_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks17_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks17_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks17_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3846722\n",
      "# reads with at least one reported alignment: 1868380 (48.57%)\n",
      "# reads that failed to align: 1909964 (49.65%)\n",
      "# reads with alignments suppressed due to -m: 68378 (1.78%)\n",
      "Reported 1868380 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks18\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks18_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks18_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks18_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks18-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks18_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11202664\n",
      "# reads with at least one reported alignment: 12564 (0.11%)\n",
      "# reads that failed to align: 11188492 (99.87%)\n",
      "# reads with alignments suppressed due to -m: 1608 (0.01%)\n",
      "Reported 12564 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks18_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks18_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks18_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks18_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks18_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11188492\n",
      "# reads with at least one reported alignment: 1313362 (11.74%)\n",
      "# reads that failed to align: 6236348 (55.74%)\n",
      "# reads with alignments suppressed due to -m: 3638782 (32.52%)\n",
      "Reported 1313362 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks18_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks18_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks18_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks18_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks18_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6236348\n",
      "# reads with at least one reported alignment: 93658 (1.50%)\n",
      "# reads that failed to align: 3568691 (57.22%)\n",
      "# reads with alignments suppressed due to -m: 2573999 (41.27%)\n",
      "Reported 93658 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks18_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks18_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks18_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks18_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks18_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3568691\n",
      "# reads with at least one reported alignment: 2445762 (68.53%)\n",
      "# reads that failed to align: 1038566 (29.10%)\n",
      "# reads with alignments suppressed due to -m: 84363 (2.36%)\n",
      "Reported 2445762 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks19\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks19_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks19_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks19_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks19-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks19_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 16810592\n",
      "# reads with at least one reported alignment: 78594 (0.47%)\n",
      "# reads that failed to align: 16725271 (99.49%)\n",
      "# reads with alignments suppressed due to -m: 6727 (0.04%)\n",
      "Reported 78594 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks19_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks19_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks19_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks19_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks19_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 16725271\n",
      "# reads with at least one reported alignment: 2102981 (12.57%)\n",
      "# reads that failed to align: 8889919 (53.15%)\n",
      "# reads with alignments suppressed due to -m: 5732371 (34.27%)\n",
      "Reported 2102981 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks19_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks19_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks19_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks19_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks19_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 8889919\n",
      "# reads with at least one reported alignment: 130522 (1.47%)\n",
      "# reads that failed to align: 5835004 (65.64%)\n",
      "# reads with alignments suppressed due to -m: 2924393 (32.90%)\n",
      "Reported 130522 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks19_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks19_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks19_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks19_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks19_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 5835004\n",
      "# reads with at least one reported alignment: 4031274 (69.09%)\n",
      "# reads that failed to align: 1657156 (28.40%)\n",
      "# reads with alignments suppressed due to -m: 146574 (2.51%)\n",
      "Reported 4031274 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks20\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks20_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks20_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks20_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks20-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks20_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11426507\n",
      "# reads with at least one reported alignment: 7901 (0.07%)\n",
      "# reads that failed to align: 11417036 (99.92%)\n",
      "# reads with alignments suppressed due to -m: 1570 (0.01%)\n",
      "Reported 7901 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks20_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks20_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks20_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks20_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks20_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 11417036\n",
      "# reads with at least one reported alignment: 1236067 (10.83%)\n",
      "# reads that failed to align: 6754351 (59.16%)\n",
      "# reads with alignments suppressed due to -m: 3426618 (30.01%)\n",
      "Reported 1236067 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks20_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks20_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks20_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks20_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks20_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6754351\n",
      "# reads with at least one reported alignment: 124597 (1.84%)\n",
      "# reads that failed to align: 2821310 (41.77%)\n",
      "# reads with alignments suppressed due to -m: 3808444 (56.39%)\n",
      "Reported 124597 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks20_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks20_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks20_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks20_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks20_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 2821310\n",
      "# reads with at least one reported alignment: 1618933 (57.38%)\n",
      "# reads that failed to align: 1151712 (40.82%)\n",
      "# reads with alignments suppressed due to -m: 50665 (1.80%)\n",
      "Reported 1618933 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks21\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks21_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks21_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks21_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks21-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks21_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 8102911\n",
      "# reads with at least one reported alignment: 16293 (0.20%)\n",
      "# reads that failed to align: 8083795 (99.76%)\n",
      "# reads with alignments suppressed due to -m: 2823 (0.03%)\n",
      "Reported 16293 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks21_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks21_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks21_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks21_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks21_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 8083795\n",
      "# reads with at least one reported alignment: 1252297 (15.49%)\n",
      "# reads that failed to align: 3202965 (39.62%)\n",
      "# reads with alignments suppressed due to -m: 3628533 (44.89%)\n",
      "Reported 1252297 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks21_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks21_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks21_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks21_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks21_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 3202965\n",
      "# reads with at least one reported alignment: 31652 (0.99%)\n",
      "# reads that failed to align: 2873018 (89.70%)\n",
      "# reads with alignments suppressed due to -m: 298295 (9.31%)\n",
      "Reported 31652 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks21_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks21_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks21_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks21_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks21_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 2873018\n",
      "# reads with at least one reported alignment: 1492743 (51.96%)\n",
      "# reads that failed to align: 1334045 (46.43%)\n",
      "# reads with alignments suppressed due to -m: 46230 (1.61%)\n",
      "Reported 1492743 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks22\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks22_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks22_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks22_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder /home/kazuki/^profilingdata/20160420_reanalyze/ks22-trimmed.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks22_ladder_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6559877\n",
      "# reads with at least one reported alignment: 25521 (0.39%)\n",
      "# reads that failed to align: 6530858 (99.56%)\n",
      "# reads with alignments suppressed due to -m: 3498 (0.05%)\n",
      "Reported 25521 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks22_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks22_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks22_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA /home/kazuki/^profilingdata/20160420_reanalyze/alignments/ladder/ks22_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks22_tRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 6530858\n",
      "# reads with at least one reported alignment: 967017 (14.81%)\n",
      "# reads that failed to align: 2748518 (42.09%)\n",
      "# reads with alignments suppressed due to -m: 2815323 (43.11%)\n",
      "Reported 967017 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks22_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks22_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks22_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/tRNA/ks22_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/tmpds/ks22_rRNA_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 2748518\n",
      "# reads with at least one reported alignment: 23443 (0.85%)\n",
      "# reads that failed to align: 2452111 (89.22%)\n",
      "# reads with alignments suppressed due to -m: 272964 (9.93%)\n",
      "Reported 23443 alignments to 1 output stream(s)\n",
      "\n",
      "bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 --un /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks22_nomatch.fastq --max /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks22_multi.fastq --al /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks22_match.fastq /home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655 /home/kazuki/^profilingdata/20160420_reanalyze/alignments/rRNA/ks22_nomatch.fastq /home/kazuki/^profilingdata/20160420_reanalyze/alignments/chr/ks22_match.SAM 1>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt 2>> /home/kazuki/^profilingdata/20160420_reanalyze/log_bowtie_ks48.txt\n",
      "# reads processed: 2452111\n",
      "# reads with at least one reported alignment: 1408232 (57.43%)\n",
      "# reads that failed to align: 997147 (40.66%)\n",
      "# reads with alignments suppressed due to -m: 46732 (1.91%)\n",
      "Reported 1408232 alignments to 1 output stream(s)\n",
      "\n",
      "2019-06-03 11:29:26.400353\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "######################################################\n",
    "from datetime import datetime\n",
    "import os\n",
    "\n",
    "inpath = filepath\n",
    "outpath = filepath\n",
    "\n",
    "log_file = inpath+'log_bowtie_ks48.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "present = str(datetime.now())\n",
    "logs.write(present+'\\n')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "path_l = outpath + 'alignments/ladder/'\n",
    "path_t = outpath + 'alignments/tRNA/'\n",
    "path_r = outpath + 'alignments/rRNA/'\n",
    "path_chr = outpath + 'alignments/chr/'\n",
    "path_temp = outpath + 'tmpds/'\n",
    "\n",
    "if not os.path.exists(path_l):\n",
    "    os.makedirs(path_l)\n",
    "if not os.path.exists(path_t):\n",
    "    os.makedirs(path_t)\n",
    "if not os.path.exists(path_r):\n",
    "    os.makedirs(path_r)\n",
    "if not os.path.exists(path_chr):\n",
    "    os.makedirs(path_chr)\n",
    "if not os.path.exists(path_temp):\n",
    "    os.makedirs(path_temp)\n",
    "    #print 'really? there is no tmpds folder?'\n",
    "\n",
    "    ###uses bowtie installed in server\n",
    "for fname in files:\n",
    "    print fname\n",
    "\n",
    "    bowtie_command = 'bowtie -v 2 -y -m 1 -a --best --strata -S -p 2 '\\\n",
    "        '--un %s%s --max %s%s --al %s%s %s %s%s %s%s 1>> %s 2>> %s'\n",
    "        # The `--strata` and `--best` options do not apply in paired-end mode.\n",
    "        # Paired-end alignments where one mate's alignment is entirely contained within the other's are considered invalid.\n",
    "\n",
    "        \n",
    "        # first, align to ladder index to subtract any ladder oligos that were cloned        \n",
    "    bowtie_ladder = bowtie_command %(\n",
    "        path_l,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_l,\n",
    "        fname+'_multi.fastq',\n",
    "        path_l,\n",
    "        fname+'_match.fastq',\n",
    "        '/home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/ladder/ladder',\n",
    "        inpath,\n",
    "        fname + bowtie_input_extension,\n",
    "        path_temp,\n",
    "        fname + '_ladder_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+fname+'\\n'+bowtie_ladder+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_ladder)        \n",
    "        \n",
    "        \n",
    "        # second, align to tRNA index\n",
    "    bowtie_tRNA = bowtie_command %(\n",
    "        path_t,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_t,\n",
    "        fname+'_multi.fastq',\n",
    "        path_t,\n",
    "        fname+'_match.fastq',\n",
    "        '/home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/Ec_tRNA',\n",
    "        path_l,\n",
    "        fname +'_nomatch.fastq',\n",
    "        path_temp,\n",
    "        fname + '_tRNA_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n'+bowtie_tRNA+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_tRNA)        \n",
    "        \n",
    "        \n",
    "        # third, align to the rRNA index\n",
    "    bowtie_rRNA = bowtie_command %(\n",
    "        path_r,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_r,\n",
    "        fname+'_multi.fastq',\n",
    "        path_r,\n",
    "        fname+'_match.fastq',\n",
    "        '/home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/rRNA_MG1655',\n",
    "        path_t,\n",
    "        fname +'_nomatch.fastq',\n",
    "        path_temp,\n",
    "        fname + '_rRNA_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n'+bowtie_rRNA+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_rRNA)         \n",
    "        \n",
    "        # finally align to the chr index\n",
    "    bowtie_chr = bowtie_command %(\n",
    "        path_chr,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_chr,\n",
    "        fname+'_multi.fastq',\n",
    "        path_chr,\n",
    "        fname+'_match.fastq',\n",
    "        '/home/kazuki/^data_analysis/^scripts/ABNG_scripts/bowtie/indexes/MG1655/e_coli_MG1655',\n",
    "        path_r,\n",
    "        fname +'_nomatch.fastq',\n",
    "        path_chr,\n",
    "        fname + '_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n'+bowtie_chr+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_chr)\n",
    "    \n",
    "\n",
    "logs = open(log_file, 'a+')\n",
    "present = str(datetime.now())\n",
    "logs.write('\\n'+present+'\\n')\n",
    "logs.close()\n",
    "    \n",
    "print \n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "######################################################\n",
    "import csv\n",
    "import pickle\n",
    "from copy import deepcopy\n",
    "\n",
    "out_path=filepath+'density/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "for f in files:\n",
    "    samgenmain_filename=filepath+'alignments/chr/'+f+'_match.SAM'\n",
    "    f_samgenmain=open(samgenmain_filename)\n",
    "    samgenmain=csv.reader(f_samgenmain,delimiter='\\t')\n",
    "\n",
    "    gff = '/home/kazuki/^data_analysis/MG1655ver2/coli.gff'\n",
    "\n",
    "    three_mapped=three_end_mapping(samgenmain,gff)\n",
    "    pickle_readcounts = out_path+f+mapping+'_readcounts.pickle'\n",
    "    with open(pickle_readcounts, 'wb') as readcounts_f:\n",
    "        pickle.dump(three_mapped,readcounts_f)\n",
    "\n",
    "    three_mapped_rpm = deepcopy(three_mapped)\n",
    "    readcounts_to_rpm(three_mapped_rpm[0])\n",
    "    readcounts_to_rpm(three_mapped_rpm[1])\n",
    "    pickle_rpm = out_path+f+mapping+'_rpm.pickle'\n",
    "    with open(pickle_rpm, 'wb') as rpm_f:\n",
    "        pickle.dump(three_mapped_rpm,rpm_f)\n",
    "        \n",
    "    density_path = out_path+f+mapping\n",
    "    writecountsf(three_mapped_rpm[0],density_path+\"_plus_\")\n",
    "    countstowig(three_mapped_rpm[0],density_path+\"_plus\")\n",
    "    writecountsf(three_mapped_rpm[1],density_path+\"_minus_\")\n",
    "    countstowig(three_mapped_rpm[1],density_path+\"_minus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# make a python-dictionary of RPKMs from each library\n",
    "import os\n",
    "import pickle\n",
    "from pandas import Series, DataFrame\n",
    "import pandas as pd\n",
    "\n",
    "in_path = filepath+'density/'\n",
    "out_path = filepath+'expression_2/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "#########################################################################################\n",
    "\n",
    "with open('/home/kazuki/^data_analysis/^Ecoli_Genome/Ecoli_Gene_TU.pickle','rb') as GT_f:\n",
    "    Gene_TU_dict = pickle.load(GT_f)\n",
    "\n",
    "for f in files:\n",
    "    expression_dict={}\n",
    "    out_pickle=out_path+f+mapping+gene_type+'_expression.pickle'\n",
    "    out_csv=out_path+f+mapping+gene_type+'_expression.csv'\n",
    "    readcounts_file = in_path+f+mapping+'_readcounts.pickle'\n",
    "    \n",
    "    with open(readcounts_file, 'rb') as rf:\n",
    "        readcounts_dict = pickle.load(rf)\n",
    "    plus_readcounts = readcounts_dict[0][readcounts_dict[0].keys()[0]]\n",
    "    minus_readcounts = readcounts_dict[1][readcounts_dict[1].keys()[0]]\n",
    "    total_read = sum(plus_readcounts)+sum(minus_readcounts)\n",
    "    \n",
    "    for key in Gene_TU_dict.keys():\n",
    "        if key in expression_dict.keys():\n",
    "            #print key +' is overlapping!!!'\n",
    "            continue\n",
    "    \n",
    "        ########           \n",
    "        if Gene_TU_dict[key][0] == '+':\n",
    "            start = int(Gene_TU_dict[key][2])-1+15\n",
    "            end = int(Gene_TU_dict[key][3])\n",
    "        if Gene_TU_dict[key][0] == '-':\n",
    "            start = int(Gene_TU_dict[key][2])-1\n",
    "            end = int(Gene_TU_dict[key][3])-15            \n",
    "        ########  \n",
    "        \n",
    "        length = end-start\n",
    "        if length <= 0:\n",
    "            #print key + ' is shorter than zero!!!'\n",
    "            continue\n",
    "        if Gene_TU_dict[key][0] == '+':\n",
    "            reads = float(sum(plus_readcounts[start:end]))\n",
    "        if Gene_TU_dict[key][0] == '-':\n",
    "            reads = float(sum(minus_readcounts[start:end]))\n",
    "        rpc = (reads/length)*3\n",
    "        rpkm = (((reads/total_read)*1000000)/length)*1000\n",
    "        expression_dict[key]=[]\n",
    "        expression_dict[key].append(reads)\n",
    "        expression_dict[key].append(rpc)\n",
    "        expression_dict[key].append(rpkm)\n",
    "    \n",
    "    with open(out_pickle, 'wb') as op:\n",
    "        pickle.dump(expression_dict,op)           \n",
    "    \n",
    "    rpkm_DataFrame = DataFrame.from_dict(expression_dict, orient='index')\n",
    "    rpkm_DataFrame.columns = ['reads','rpc','rpkm']\n",
    "    rpkm_DataFrame.to_csv(out_csv) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "good\n"
     ]
    }
   ],
   "source": [
    "print \"good\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "######################################\n",
    "########### RPKMs for 3UTR ###########\n",
    "######################################\n",
    "\n",
    "import os\n",
    "import pickle\n",
    "from pandas import Series, DataFrame\n",
    "import pandas as pd\n",
    "\n",
    "in_path = filepath+'density/'\n",
    "out_path = filepath+'expression_3UTR/' ##################\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "#########################################################################################\n",
    "\n",
    "with open('/home/kazuki/^data_analysis/^Ecoli_Genome/Ecoli_Gene_TU.pickle','rb') as GT_f:\n",
    "    Gene_TU_dict = pickle.load(GT_f)\n",
    "\n",
    "for f in files:\n",
    "    expression_dict={}\n",
    "    out_pickle=out_path+f+mapping+gene_type+'_expression.pickle'\n",
    "    out_csv=out_path+f+mapping+gene_type+'_expression.csv'\n",
    "    readcounts_file = in_path+f+mapping+'_readcounts.pickle'\n",
    "    \n",
    "    with open(readcounts_file, 'rb') as rf:\n",
    "        readcounts_dict = pickle.load(rf)\n",
    "    plus_readcounts = readcounts_dict[0][readcounts_dict[0].keys()[0]]\n",
    "    minus_readcounts = readcounts_dict[1][readcounts_dict[1].keys()[0]]\n",
    "    total_read = sum(plus_readcounts)+sum(minus_readcounts)\n",
    "    \n",
    "    for key in Gene_TU_dict.keys():\n",
    "        if key in expression_dict.keys():\n",
    "            #print key +' is overlapping!!!'\n",
    "            continue\n",
    "            \n",
    "            \n",
    "        ########\n",
    "        if type(Gene_TU_dict[key][4]) != str:\n",
    "            continue\n",
    "\n",
    "        TheLASTGene = Gene_TU_dict[key][4].split('-')[-1]\n",
    "        if key != TheLASTGene:\n",
    "            continue\n",
    "            \n",
    "        if Gene_TU_dict[key][0] == '+':\n",
    "            start = int(Gene_TU_dict[key][3])-1+15\n",
    "            end = int(Gene_TU_dict[key][6])\n",
    "        if Gene_TU_dict[key][0] == '-':\n",
    "            start = int(Gene_TU_dict[key][5])-1\n",
    "            end = int(Gene_TU_dict[key][2])-15\n",
    "        ########\n",
    "        \n",
    "        \n",
    "        length = end-start\n",
    "        if length <= 0:\n",
    "            #print key + ' is shorter than zero!!!'\n",
    "            continue\n",
    "        if Gene_TU_dict[key][0] == '+':\n",
    "            reads = float(sum(plus_readcounts[start:end]))\n",
    "        if Gene_TU_dict[key][0] == '-':\n",
    "            reads = float(sum(minus_readcounts[start:end]))\n",
    "        rpc = (reads/length)*3\n",
    "        rpkm = (((reads/total_read)*1000000)/length)*1000\n",
    "        expression_dict[key]=[]\n",
    "        expression_dict[key].append(reads)\n",
    "        expression_dict[key].append(rpc)\n",
    "        expression_dict[key].append(rpkm)\n",
    "    \n",
    "    with open(out_pickle, 'wb') as op:\n",
    "        pickle.dump(expression_dict,op)           \n",
    "    \n",
    "    rpkm_DataFrame = DataFrame.from_dict(expression_dict, orient='index')\n",
    "    rpkm_DataFrame.columns = ['reads','rpc','rpkm']\n",
    "    rpkm_DataFrame.to_csv(out_csv) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "good\n"
     ]
    }
   ],
   "source": [
    "print \"good\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
